### PR TITLE
[BREAKING Responsive] Start swapping components

### PR DESCRIPTION
**Pull request by damassi**
Supersedes: https://github.com/artsy/reaction/pull/1379
Builds on: https://github.com/artsy/react-responsive-media
Discussion in: https://github.com/artsy/reaction/issues/1367

Closes https://github.com/artsy/reaction/pull/1379
Closes https://github.com/artsy/reaction/issues/1367

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 5213babf822aa70f16c016d8b9c9a4d09a4a1cdc

https://deploy-preview-1466--artsy-reaction.netlify.com

File: src/Apps/Artwork/Components/ImageCarousel.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import styled from "styled-components"
 
 import Icon from "Components/Icon"
 import { Arrow } from "Styleguide/Elements/Arrow"
-import { Responsive } from "Utils/Responsive"
+import { Media } from "Utils/Responsive"
```
Comment (damassi) [inline src/Apps/Artwork/Components/ImageCarousel.tsx]: @alloy - would you mind reviewing this file's refactor (ImageCarousel) and seeing if this is the right approach? There are conditional elements throughout and I was able to (somewhat) cleanly refactor them into the new approach but wondering if there might be a better way or if there's something we could do at the lib level in order to avoid having to execute conditional functions within each breakpoint and pass strings. It pretty much seems ok, but we're having to hoist elements out of the main `render` in order to keep things reasonable which would be nice not to have to do. I could be missing something!

File: src/Apps/Artwork/Components/ImageCarousel.tsx [hunk 2]
```diff
@@ -177,74 +177,116 @@ export class ImageCarousel extends React.Component<
...
+
+  render() {
+    return (
+      <Container>
+        <Media at="xs">
+          {this.renderImageArea("xs")}
+          {this.renderControlsContainer("xs")}
+        </Media>
+
+        <Media greaterThan="xs">
```
Comment (damassi) [inline src/Apps/Artwork/Components/ImageCarousel.tsx]: (I'm gonna refactor all of these `greaterThan='xs'` at once in another commit)

File: src/Apps/Artwork/Components/ImageCarousel.tsx [hunk 3]
```diff
@@ -177,74 +177,116 @@ export class ImageCarousel extends React.Component<
...
+          {this.renderImageArea()}
+          {this.renderControlsContainer()}
+        </Media>
+      </Container>
```
Comment (damassi) [inline src/Apps/Artwork/Components/ImageCarousel.tsx]: (I'm gonna refactor all of these greaterThan='xs' at once in another commit)

File: src/Apps/Artist/Routes/AuctionResults/ArtistAuctionResultItem.tsx [hunk 4]
```diff
@@ -32,36 +32,19 @@ export const ArtistAuctionResultItem: SFC<Props> = props => {
...
+                mediator={mediator}
+                user={user}
+              />
+            </Media>
+            <Media between={["sm", "xl"]}>
+              <SmallAuctionItem {...props} mediator={mediator} user={user} />
+            </Media>
+            <Media at="xl">
+              <LargeAuctionItem {...props} mediator={mediator} user={user} />
+            </Media>
```
[the remark this answers is not in the document]
Comment (alloy) [reply src/Apps/Artist/Routes/AuctionResults/ArtistAuctionResultItem.tsx]: Reminder ^

File: src/Apps/Order/Routes/__tests__/Payment.test.tsx [hunk 5]
```diff
@@ -504,8 +513,9 @@ describe("Payment", () => {
       paymentRoute.find("Button").simulate("click")
       expect(stripeMock.createToken).toBeCalled()
     })
-    it("allows a missing state/province if the selected country is not US or Canada", () => {
-      const paymentRoute = mount(<PaymentRoute {...testProps} />)
+
+    it.only("allows a missing state/province if the selected country is not US or Canada", () => {
```
Comment (damassi) [inline src/Apps/Order/Routes/__tests__/Payment.test.tsx]: @alloy - would you mind pulling down this branch and seeing if you can fix the error below? I'm a bit stumped. 

```
yarn jest src/Apps/Order/Routes/__tests__/Payment.test.tsx --watch --verbose false
```

There are quite a few strange test errors throughout and I suspect if the below is fixable then the rest will mostly resolve (after being wrapped in `MockBoot`).

File: src/setup_jest.ts [hunk 6]
```diff
@@ -13,7 +13,7 @@ jest.mock("react-sizeme", () => jest.fn(c => d => d))
  * We want each test to have assertions, otherwise it’s too easy to write async
  * tests that never end up making any, leading to false positives.
  */
-beforeEach(() => expect.hasAssertions())
+afterEach(() => expect.hasAssertions())
```
Comment (damassi) [inline src/setup_jest.ts]: This was consuming usable errors. If set to `after` it will still check the test for assertions yet errors within code will print just above.

File: src/Styleguide/Components/ArtistBio.tsx [hunk 7]
```diff
@@ -18,26 +18,25 @@ export const MAX_CHARS = {
...
-@track({ context_module: Schema.ContextModule.ArtistBio })
+@track({
+  context_module: Schema.ContextModule.ArtistBio,
+})
 export class ArtistBio extends React.Component<ArtistBioProps> {
+  renderReadMore = size => {
+    return (
+      <ReadMore
+        onReadMoreClicked={this.props.onReadMoreClicked}
+        maxChars={size === "xs" ? MAX_CHARS.xs : MAX_CHARS.default}
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/Styleguide/Components/ArtistBio.tsx]: I would rather not update existing behavior in this PR but feel free to file an issue.

File: src/Styleguide/Utils/BreakpointVisualizer.tsx [hunk 8]
```diff
@@ -1,92 +1,36 @@
...
+  <React.Fragment>
+    {typeof window !== "undefined" &&
+      window.navigator &&
+      !window.navigator.userAgent.match(/Chromatic/) && (
+        <StyledBox top={0} left={0} py={1} px={2}>
+          <BreakpointText breakpoint="xs" />
+          <BreakpointText breakpoint="sm" />
+          <BreakpointText breakpoint="md" />
+          <BreakpointText breakpoint="lg" />
+          <BreakpointText breakpoint="xl" />
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/Styleguide/Utils/BreakpointVisualizer.tsx]: Can you elaborate a bit?

File: src/Apps/Artwork/Components/ImageCarousel.tsx (hunk 3 above)
Comment (alloy) [reply src/Apps/Artwork/Components/ImageCarousel.tsx]: Not sure I understand what you were going to refactor here, can you elaborate?

Comment (damassi) [reply src/Apps/Artwork/Components/ImageCarousel.tsx]: `greaterThan` -> `greaterThanOrEqual` (but i think this is already done)

Comment (damassi) [reply src/Apps/Artwork/Components/ImageCarousel.tsx]: (Actually, gonna push that now)

File: src/Styleguide/Components/ArtistBio.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (alloy) [reply src/Styleguide/Components/ArtistBio.tsx]: Good point 👍

File: src/Styleguide/Utils/BreakpointVisualizer.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (alloy) [reply src/Styleguide/Utils/BreakpointVisualizer.tsx]: I was musing out loud about a way to tell `Media` to _always_ render all variants regardless of dynamic client-side breakpoint determination.

[the remark this answers is not in the document]
Comment (alloy) [reply src/Styleguide/Utils/BreakpointVisualizer.tsx]: (Nothing that is actually needed right now)

File: src/Apps/Artwork/Components/ImageCarousel.tsx (hunk 3 above)
Comment (damassi) [reply src/Apps/Artwork/Components/ImageCarousel.tsx]: done!

Comment (alloy) [reply src/Apps/Artwork/Components/ImageCarousel.tsx]: Ah gotcha. I actually think that when next to a `at="xs"` one it reads nice to use `greaterThan="xs"` but in standalone cases I agree that `greaterThanOrEqual` reads better. Anyways, no need to keep going back and forth on this right now.

Comment (damassi) [reply src/Apps/Artwork/Components/ImageCarousel.tsx]: In that case I'm gonna pop off that last commit because I think it reads better alongside too.

Comment (alloy) [reply src/Apps/Artwork/Components/ImageCarousel.tsx]: Hah, nice 👍

File: src/Apps/Order/Routes/__tests__/Payment.test.tsx [hunk 9]
```diff
@@ -62,8 +65,16 @@ const fillAddressForm = (component: any, address: Address) => {
...
+  const getWrapper = (props, breakpoint: Breakpoint = "xs") => {
+    return mount(
+      <MockBoot breakpoint={breakpoint}>
+        <PaymentRoute {...props} />
+      </MockBoot>
+    )
+  }
+
   beforeEach(() => {
-    console.error = jest.fn() // Silences component logging.
```
Comment (alloy) [inline src/Apps/Order/Routes/__tests__/Payment.test.tsx]: @damassi This was masking actual errors, so removed it.

* At first a render error occurred that failed rendering due to trying to pass the `hidePostalCode` prop that was being passed to a mocked `CardElement` that was returning a `div` element.
* Then the actual issue showed up, which is `Warning: Can't call setState (or forceUpdate) on an unmounted component.`

Looking into that issue is up next.

Comment (damassi) [reply src/Apps/Order/Routes/__tests__/Payment.test.tsx]: Ugh! just like we talked about the other day. I suspect there might be a few other examples of `console.log` / `console.error` being consumed around reaction, might be good to grep other tests.

File: src/Apps/Artwork/Components/ImageCarousel.tsx (hunk 1 above)
Comment (damassi) [reply src/Apps/Artwork/Components/ImageCarousel.tsx]: I'm going to copy this file's changes into my current branch

**Comment by alloy** [comment]
This is mostly done, just needs a rebase but I was having a bit of a hard time with the ImageCarousel one and figured that you (@damassi) still have that context fresh in your mind. Would you mind doing the rebase?

**Comment by damassi** [comment]
@alloy - yup, sure thing; one sec...

File: src/Apps/Artwork/Components/__tests__/ArtworkDetails.test.tsx [hunk 10]
```diff
@@ -17,6 +18,7 @@ describe("ArtworkDetails", () => {
           }
         }
       `,
+      wrapper: n => <MockBoot breakpoint="xs">{n}</MockBoot>,
```
Comment (damassi) [inline src/Apps/Artwork/Components/__tests__/ArtworkDetails.test.tsx]: We can do this in another PR but should pop off the `breakpoint='xs'` and defer to defaults

**Comment by damassi** [comment]
Alright, lets do this 👍 

💯 💯 💯
